utils/src/treev2/quickPickWizard/experiences/subscriptionExperience.ts

```diff
@@ -0,0 +1,32 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as types from '../../../../index';
+import * as vscode from 'vscode';
+import { AzureWizard } from '../../../wizard/AzureWizard';
+import { QuickPickAzureSubscriptionStep } from '../quickPickAzureResource/QuickPickAzureSubscriptionStep';
+import { ApplicationSubscription, PickSubscriptionWizardContext } from '../../../../hostapi.v2';
+import { ResourceGroupsItem } from '../quickPickAzureResource/tempTypes';
+import { NoResourceFoundError } from '../../../errors';
+
+export async function subscriptionExperience(context: types.IActionContext, tdp: vscode.TreeDataProvider<ResourceGroupsItem>): Promise<ApplicationSubscription> {
```

Comment:
Should `context` be of a type that folds in `PickSubscriptionWizardContext`? Or, if the intent is that the caller not provide any "pre-filled" context that would otherwise bypass wizard steps, then it seems like the wizard context should be an object that's independent of the passed context.  Otherwise, the context will end up with a `subscription` property set as a side-effect, which may not be expected.

This might also work:

```TypeScript
const wizardContext = { ...context } as PickSubscriptionWizardContext;
```

Reply:
(And the question of the other contexts in the "experience" functions.)